### PR TITLE
Allow Docker image cache expiry to be specified from the command line

### LINKED ISSUE #47: Allow Docker Image cache interval to be configured as a command line option
More convenient to tune the tradeoff between performance and image freshness.

### PR DESCRIPTION
**Pull request by fonglh**
Also change poll interval configuration to use `ActiveSupport::Configurable`.

Fixes #47.

### CODE REVIEW

**Comment by allenwq** [comment]
@fonglh  I would prefer to have a global config instead of making each class configurable, that's more cleaner and maintainable to me:
```ruby
module Coursemology::Evaluator
  include ActiveSupport::Configurable
end
```

Then we can access the config across the gem by `Evaluator.config.xxx`

What do you think ?  despite of this all look good to me.

**Comment by allenwq** [comment]
LGTM

File: README.md [hunk 1]
```diff
@@ -31,3 +31,26 @@ full explanation.
 
 3. Start the evaluator using the Procfile. You can use [foreman](https://github.com/ddollar/foreman)
    or any similar tool to generate system scripts for boot.
+
+### Command Line Options
+
+#### Compulsory Options
+
+1. `--host`: Coursemology host to connect to
+1. `--api-user-email`: User with autograder flag set
```
Comment (lowjoel) [inline README.md]: I typically number the options properly, although it's lazier to do this...

**Comment by lowjoel** [comment]
Calling it cache expiry is a misnomer. A better name might me image lifetime or something to that effect. 

This cannot be too long because then different evaluators will have different images and since job assignment is random sometimes a job might evaluate to succeed in one image and fail in another. Take note.

**Comment by fonglh** [comment]
Ok I'll use a different name.

I'm not expecting our images to change often, and if they do, we can restart the evaluator and that'll clear the cache right?

**Comment by lowjoel** [comment]
you can, but you'll need to find a way to script it. also need to figure out how and when to manage changes to the images repo.

**Comment by fonglh** [comment]
It'll be manual for now. We'll focus on supporting CS1010.

That's why I think #45 will be useful if other people really do want to run their own.

**Comment by lowjoel** [comment]
@allenwq I'm OK with this change in general, if you have no concerns about the implementation, please merge.

**Comment by allenwq** [comment]
@lowjoel I have no issue, @fonglh discussed this with me offline.

**Pull request merged**